x-pack/plugins/stack_connectors/server/connector_types/stack/opsgenie/schema.ts

```diff
@@ -0,0 +1,126 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { schema } from '@kbn/config-schema';
+import { i18n } from '@kbn/i18n';
+
+export const ConfigSchema = schema.object({
+  apiUrl: schema.string(),
+});
+
+export const SecretsSchema = schema.object({
+  apiKey: schema.string(),
+});
+
+const responderTypes = schema.oneOf([
+  schema.literal('team'),
+  schema.literal('user'),
+  schema.literal('escalation'),
+  schema.literal('schedule'),
+]);
+
+const validateDetails = (details: Record<string, string>): string | void => {
+  let totalChars = 0;
+
+  for (const [key, value] of Object.entries(details)) {
+    totalChars += key.length + value.length;
+
+    if (totalChars > 8000) {
+      return i18n.translate('xpack.stackConnectors.opsgenie.invalidDetails', {
+        defaultMessage: 'details field character count exceeds the 8000 limit',
+      });
+    }
+  }
+};
```

Comment:
According to the docs [here](https://support.atlassian.com/opsgenie/docs/alert-fields/) we should only validate the values and not the whole JSON string. The `Extra Properties` is the `details`. You can test it if you open the Network Tab in your dev tools. The API docs are misleading. I though the same at first.

Also, if we had to validate the whole JSON string I think it is better to do

```suggestion
const validateDetails = (details: Record<string, string>): string | void => {
  let totalChars = 0;
  try {
    JSON.stringify(details).length
  } catch (e) { // throw error about JSON stringify }
  
  if (totalChars > 8000) {
      return i18n.translate('xpack.stackConnectors.opsgenie.invalidDetails', {
        defaultMessage: 'details field character count exceeds the 8000 limit',
      });
    }
};
```
because we need to validate `{`, `,`, and `}`

Reply:
Great catch!